src/glmfit.jl

```diff
@@ -115,7 +115,23 @@ function confint(obj::AbstractGLM, level::Real)
 end
 confint(obj::AbstractGLM) = confint(obj, 0.95)
 
-deviance(m::AbstractGLM)  = deviance(m.rr)
+deviance(m::AbstractGLM) = deviance(m.rr)
+
+function loglikelihood(m::AbstractGLM)
```

Comment:
The squared deviance residuals returned by the `devresid` method are defined as `deviance - null deviance` for each observation.  The calculation of the deviance itself should remain as it is because it is the deviance that you need to evaluate most frequently. The current methods are fast and only involve a single pass over the observations. I don't think there is a problem with evaluating the loglikelihood in a separate loop because that calculation is needed much less frequently.